src/boot/efi/boot.c

```diff
@@ -2642,6 +2643,16 @@ static void config_load_all_entries(
         config_default_entry_select(config);
 }
 
+EFI_STATUS discover_root_dir(EFI_LOADED_IMAGE_PROTOCOL *loaded_image, EFI_FILE **ret_dir) {
+        EFI_STATUS err;
```

Comment:
As long as this function is this trivial it's imho nicer to just return the results directly and do away with this var.